### PR TITLE
transmission: fix compile with mbedtls3.X

### PR DESCRIPTION
**Pull request by orangepizza**
Signed-off-by: Seo Suchan <tjtncks@gmail.com>

Maintainer: Daniel Golle <daniel@makrotopia.org> / @dangowrt (find it by checking history of the package Makefile)
Compile tested: x64, libvirt, master
Run tested:  x64, libvirt, master, downloading a torrent

Description: insert a patch to make it use renamed function names when complied againest  mbedtls3.x, while update to most resent version too.

Fixes: https://github.com/openwrt/packages/issues/24120

### CODE REVIEW

**Comment by Ansuel** [comment]
@orangepizza Can you split this into 2 commit (one for patch and one for release bump?) and wrap to 80 columns? 

Also wonder if you proposed this upstream? Also I notice some typo in the commit description.

**Comment by orangepizza** [comment]
Made a upstream pr https://github.com/transmission/transmission/pull/6822

**Comment by systemcrash** [comment]
"compile" surely :)

**Comment by BKPepe** [comment]
It seems like upstream developers are interested in the fix and we should wait a little bit before merging this here. :)

**Comment by BKPepe** [comment]
1. Fixed typo in commit subject
2. Improved commit description
3. Backported patch to have the latest changes, which you submitted to upstream and it now, it is possible to apply the patch via git am.

**Comment by orangepizza** [comment]
>     3. Backported patch to have the latest changes, which you submitted to upstream and it now, it is possible to apply the patch via git am.

this patch is what really tested and upstream PR is just cherry pick of this, so it'd be upstream PR that will be modified

File: net/transmission/patches/010-temp-mbedtls3-compile.patch
Comment (BKPepe) [inline net/transmission/patches/010-temp-mbedtls3-compile.patch:4]: This commit subject is not what is going to be applied in upstream. This patch should be equal to that.

1. https://github.com/transmission/transmission/pull/6822
2. https://github.com/transmission/transmission/pull/6823

The approach for is is to add to the end of the url .patch, so it would look like this https://github.com/transmission/transmission/pull/6823.patch

**Review by BKPepe — CHANGES_REQUESTED** [review]

**Comment by BKPepe** [comment]
Your commit description was too long. It should have max 80 chars per row. Thats why I edited it, working on it since this was your 1st contribution here.

File: net/transmission/patches/010-temp-mbedtls3-compile.patch
Comment (BKPepe) [inline net/transmission/patches/010-temp-mbedtls3-compile.patch:1]: Why you are doing this change? 2From? It was ready to be merged, I was waiting for CI/CD... 
Please, don't do those changes.

Comment (BKPepe) [reply net/transmission/patches/010-temp-mbedtls3-compile.patch]: Reverted your change.

Comment (orangepizza) [reply net/transmission/patches/010-temp-mbedtls3-compile.patch]: for some reason refresh made that

**Pull request merged**
